### PR TITLE
Site notification

### LINKED ISSUE #68: Site notification feature
On main page:
- Allow to see a list of notifications (such as reminders set by user)
- Notifications can be removed from the list if desired
- Minimizable

### PR DESCRIPTION
**Pull request by MortarDefender**
- adds two new URLs get a notification and seen notification
- get notification -> return all the notifications that the user hasn't
  seen
- seen notification let the server know that the user has seen this
  notification and do not need to send them again the next time
- adds manager to reminder class

- fix: #68

### CODE REVIEW

**Comment by MortarDefender** [comment]
image

**Comment by MortarDefender** [comment]
image

File: users/static/site_notification/notification.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+				line += "				<div class='cls' onclick=remove_from_list('" + String(element.id) + "')><i class='fa fa-times-circle' style='color: red;'></i></div>";
+				line += "		</div>";
+
+			$(id).append(line);
+		});
+	})
+}
+
+function open_notification_bar() {
+	console.log(" in open");
```
Comment (RonHachmon) [inline users/static/site_notification/notification.js]: Please delete all debug notes

File: reminders/views.py [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+            participant_id__user_id=user,
+            seen_time__isnull=True
+        ).values('id', 'message')
+    )
+
+    return JsonResponse(user_notifications, safe=False)
+
+
+@login_required(login_url="login")
+def seen_notification(request, pk):
```
Comment (RonHachmon) [inline reminders/views.py]: can you please rename pk to somthing like notification_id?

File: main/utilities.py [hunk 3]
```diff
@@ -10,6 +12,7 @@ def send_mail_notification(title, message, receiver_email):
 def create_notification(message, receiver_id):
     """ create a notification with the receiver id the message given using the site notification """
     print(f"notification -> {message}  -  {receiver_id}")  # for debug only
```
Comment (RonHachmon) [inline main/utilities.py]: is this debug line still necessary?

File: users/static/site_notification/notification.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		$(id).empty();
+
+		let length = data.length;
+		if (length === 0) {
+			length = "";
+		}
+
+		$("#notification-badge").text(length);  // change the icon to the amount of notification
+
+		data.forEach((element, index) => {
```
Comment (RonHachmon) [inline users/static/site_notification/notification.js:24]: can you please add a message when there no new notification, indicating it.

**Review by RonHachmon — CHANGES_REQUESTED** [review]

File: users/static/site_notification/notification.js (hunk 4 above)
Comment (MortarDefender) [reply users/static/site_notification/notification.js:24]: done. thanks 😄

File: main/utilities.py (hunk 3 above)
Comment (MortarDefender) [reply main/utilities.py]: done. thanks 😄

File: reminders/views.py (hunk 2 above)
Comment (MortarDefender) [reply reminders/views.py]: done. thanks 😄

File: users/static/site_notification/notification.js (hunk 1 above)
Comment (MortarDefender) [reply users/static/site_notification/notification.js]: done. thanks 😄

**Review by RonHachmon — APPROVED** [review]
Great work 😬

**Review by YaelD — COMMENTED** [review]
Looks great! 👍

**Review by YaelD — APPROVED** [review]

**Review by kobihk — APPROVED** [review]

**Pull request merged**